packages/@aws-cdk/aws-ecr-assets/test/image-asset.test.ts

```diff
@@ -271,11 +271,33 @@ describe('image asset', () => {
     // When the deprecated property is removed source code, this block can be dropped.
 
     testFutureBehavior('docker directory is staged without files specified in .dockerignore', flags, App, (app) => {
-      testDockerDirectoryIsStagedWithoutFilesSpecifiedInDockerignore(app);
+      testDockerDirectoryIsStagedWithoutFilesSpecifiedInDockerignore(app, false);
+    });
+
+    testFutureBehavior('docker directory is staged without files specified in Dockerfile.dockerignore', flags, App, (app) => {
+      testDockerDirectoryIsStagedWithoutFilesSpecifiedInDockerignore(app, true);
     });
 
     testFutureBehavior('docker directory is staged without files specified in .dockerignore with IgnoreMode.GLOB', flags, App, (app) => {
-      testDockerDirectoryIsStagedWithoutFilesSpecifiedInDockerignore(app, IgnoreMode.GLOB);
+      testDockerDirectoryIsStagedWithoutFilesSpecifiedInDockerignore(app, false, IgnoreMode.GLOB);
+    });
+
+    testFutureBehavior('docker directory is staged without files specified in Dockderfile.dockerignore with IgnoreMode.GLOB', flags, App, (app) => {
+      testDockerDirectoryIsStagedWithoutFilesSpecifiedInDockerignore(app, true, IgnoreMode.GLOB);
+    });
+  });
+
+  describe('docker file ignores with multiple custom images', () => {
+    testFutureBehavior('Dockerfile w/ .dockerignore', flags, App, (app) => {
+      testDockerDirectoryIsStagedByCustomIgnoreFiles(app, 'Dockerfile', ['.dockerignore', 'default.txt', 'index.py']);
+    });
+
+    testFutureBehavior('DockerfileA w/ DockerfileA.dockerignore', flags, App, (app) => {
+      testDockerDirectoryIsStagedByCustomIgnoreFiles(app, 'DockerfileA', ['DockerfileA.dockerignore', 'a.txt', 'index.py']);
+    });
+
+    testFutureBehavior('DockerfileB w/ DockerfileB.dockerignore', flags, App, (app) => {
+      testDockerDirectoryIsStagedByCustomIgnoreFiles(app, 'DockerfileB', ['DockerfileB.dockerignore', 'b.txt', 'index.py']);
```

Comment:
These three tests seem to test the exact same thing...why is there not just one of these?

Reply:
this was just to verify that the different config scenarios do not contaminate each other, meaning that each ignore file is respected individually by checking that the expected files exist and the others do not. so it's checking that .dockerignore is not affecting DockerfileA.dockerignore, and vice versa. it will hopefully detect any unintended changes to this behavior in the future. the DockerfileB test is just another permutation ensuring the custom dockerignore files do not affect each other.

Reply:
at most we need two of these. Could we remove the first test here, and leave `A` and `B`?

Reply:
yeah, works for me, that one is the most redundant.  didn't mean to take so long to respond.  a3b99ca has the `.dockerignore` test removed.